packages/react-i18n/README.md

```diff
@@ -123,7 +123,10 @@ export default withI18n()(NotFound);
 The provided `i18n` object exposes many useful methods for internationalizing your apps. You can see the full details in the [`i18n` source file](https://github.com/Shopify/quilt/blob/master/packages/react-i18n/src/i18n.ts), but you will commonly need the following:
 
 - `formatNumber()`: formats a number according to the locale. You can optionally pass an `as` option to format the number as a currency or percentage; in the case of currency, the `defaultCurrency` supplied to the i18n `I18nContext.Provider` component will be used where no custom currency code is passed.
-- `formatCurrency()`: formats a number as a currency according to the locale. Convenience function that simply _auto-assigns_ the `as` option to `currency` and calls `formatNumber()`.
+- `formatCurrency()`: formats a number as a currency according to the locale. Its behaviour depends on the `form:` option.
+  - if `form: 'short'` is given, then a possibly-ambiguous short form is used, consisting of the bare symbol if the currency has a symbol, or the ISO 4217 code if there is no symbol for that currency.  Examples:  `CHF 1,25`, `€ 1,25 EUR`, `OMR 1.250`, `$ 1.25 USD`
+  - if `form: 'explicit'` is given, then the result will be the same as for `short`, but will append the ISO 4217 code if it is not already present
+  - if `form:` is not given, then behaviour reverts to the legacy (deprecated) `formatCurrency()`, which is a convenience function that simply _auto-assigns_ the `as` option to `currency` and calls `formatNumber()`.  Note that this will resemble `form: 'short'`, but will sometimes extend the symbol with extra information depending on the browser's implementation of `Intl.NumberFormat` and the locale in use.  For example, `formatCurrency(1.25, {currency: 'CAD'})` may return `$ 1.25`, or it might return `CA$ 1.25`.
```

Comment:
> then behaviour reverts to the legacy (deprecated)

Is this _really_ being deprecated? It's used in quite a few places so we'll have to be careful with how we deprecate it if we choose to. 

> but will sometimes extend the symbol with extra information depending on the browser's implementation of `Intl.NumberFormat` and the locale in use.

Do browser implementations dictate this? I thought it was just locales passed in?

Eg. 

```
console.log(new Intl.NumberFormat('en-US', { style: 'currency', currency: 'CAD' }).format(123456.789));
// > "CA$123,456.79"

console.log(new Intl.NumberFormat('en-CA', { style: 'currency', currency: 'CAD' }).format(123456.789));
// > "$123,456.79"
```

Reply:
The ECMAScript standard says that a currency symbol may be provided for a given combination of (currency, locale), but leaves what symbol is provided (or, even, whether a symbol is given at all, or whether the ISO 4217 code is used) as "implementation dependent":
https://www.ecma-international.org/ecma-402/6.0/index.html#sec-properties-of-intl-numberformat-constructor

Firefox, Safari and Chrome all seem to use the same implementation, but there's no guarantee that that will be the case for all browsers.

Reply:
Regarding deprecation of the legacy `formatCurrency()`... we should bring @paulstairmand and @winterweix into that conversation.